opennms-config-model/src/main/java/org/opennms/netmgt/config/provisiond/ProvisiondConfiguration.java

```diff
@@ -33,44 +33,33 @@
 import java.util.List;
 import java.util.Objects;
 
-import javax.xml.bind.annotation.XmlAccessType;
-import javax.xml.bind.annotation.XmlAccessorType;
-import javax.xml.bind.annotation.XmlAttribute;
-import javax.xml.bind.annotation.XmlElement;
-import javax.xml.bind.annotation.XmlRootElement;
-
-import org.opennms.core.xml.ValidateUsing;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import org.opennms.netmgt.config.utils.ConfigUtils;
 
 /**
  * Behavior configuration for the Provisioner Daemon
  */
-@XmlRootElement(name = "provisiond-configuration")
-@XmlAccessorType(XmlAccessType.FIELD)
-@ValidateUsing("provisiond-configuration.xsd")
 public class ProvisiondConfiguration implements Serializable {
     private static final long serialVersionUID = 2L;
 
     private static final String DEFAULT_REQUISITION_DIR = "${install.dir}/etc/imports";
     private static final String DEFAULT_FOREIGN_SOURCE_DIR = "${install.dir}/etc/foreign-sources";
 
-    @XmlAttribute(name = "importThreads")
-    private Long m_importThreads;
+    @JsonProperty("importThreads")
```

Comment:
Our thoughts were those:

- new field means changed XSD => we create a new version of the XSD  (XSD versions are frozen and cannot be changed)
- import new xsd version via liquibase
- at the same time we add that attribute to the java class

Reply:
We have for properties the logic in place to express validation directly in the liquibase. We could consider doing the same for xml since both are operating on OpenApi json.